collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -116,6 +123,18 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main_menu);
         initToolbar();
 
+        // Enable TLS v1.2 for Android 4.2.2
+        try {
+            ProviderInstaller.installIfNeeded(getApplicationContext());
+            SSLContext sslContext;
+            sslContext = SSLContext.getInstance("TLSv1.2");
```

Comment:
This seems to force TLSv1.2. What happens if a server is using TLS v1.1 or TLS v1.3? It seems like a better approach (dunno if this possible) is let the server chose?

Reply:
Seems like this code creates an `SSLContext` and `SSLEngine` but then doesn't do anything with them (as far as I can see). I don't think it will force any TLS version for the app but what is the purpose of it @trebla64?

Reply:
@seadowg I copied the code from stackoverflow.  I'm not sure what the sslContext is doing or if it is just dead code.  From looking at the medium article you posted it looks like the `installIfNeeded` call is the important one.  I will try testing at some point to see if the `sslContext` coded can be removed.